source/extensions/http/header_validators/envoy_default/header_validator_factory.cc

```diff
@@ -0,0 +1,36 @@
+#include "source/extensions/http/header_validators/envoy_default/header_validator_factory.h"
+
+#include "source/extensions/http/header_validators/envoy_default/http1_header_validator.h"
+#include "source/extensions/http/header_validators/envoy_default/http2_header_validator.h"
+#include "source/extensions/http/header_validators/envoy_default/null_header_validator.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Http {
+namespace HeaderValidators {
+namespace EnvoyDefault {
+
+using ::envoy::extensions::http::header_validators::envoy_default::v3::HeaderValidatorConfig;
+using ::Envoy::Http::Protocol;
+
+HeaderValidatorFactory::HeaderValidatorFactory(const HeaderValidatorConfig& config)
+    : config_(config) {}
+
+::Envoy::Http::HeaderValidatorPtr
+HeaderValidatorFactory::create(Protocol protocol, StreamInfo::StreamInfo& stream_info) {
+  switch (protocol) {
+  case Protocol::Http2:
+    return std::make_unique<Http2HeaderValidator>(config_, protocol, stream_info);
+  case Protocol::Http11:
+  case Protocol::Http10:
+    return std::make_unique<Http1HeaderValidator>(config_, protocol, stream_info);
+  default:
```

Comment:
We should remove the `default` clause. It will cause compiler failure if we ever add enum value that is not covered by any `case`. And add RELEASE_ASSERT after the switch in case we ended up getting some out of range value as a `protocol` due to a bug.